pip/flatpak-pip-generator

```diff
@@ -78,6 +78,24 @@ def get_pypi_url(name: str, filename: str) -> str:
                     return source['url']
         raise Exception('Failed to extract url from {}'.format(url))
 
+def get_pypi_changelog(name: str) -> str:
+    url = 'https://pypi.org/pypi/{}/json'.format(name)
+    with urllib.request.urlopen(url) as response:
+        body = json.loads(response.read().decode('utf-8'))
+        urls = {k.casefold(): v for k, v in body['info']['project_urls'].items()}
+        # Keys as recognized by pypi: 
+        # https://github.com/pypi/warehouse/blob/main/warehouse/templates/packaging/detail.html#L24
+        search_keys = {"changelog", "change log", "changes", "release notes", "news", "what's new", "history"}
+        keys = search_keys.intersection(urls.keys())
+        if len(keys) == 1:
+            return urls[keys.pop()]
+        elif len(keys) > 1:
+            print(f"Warning: Got multiple potential keys for the changelog, picking one at random from {keys}.")
+            return urls[keys.pop()]
```

Comment:
I'd just combine these two cases and pick the first one always. If it's wrong it needs to be manually edited anyways, the warning doesn't particularly mean that it was wrong either.

Reply:
I don't have any data on how many projects might have values for more than one of those keys or if the keys may actually have different semantic meanings. (In some projects on Github I definitely saw differences in changlog and release notes.)
Therefore my intention was to give the users a hint that they might not get the best result or then one they are expecting. It could be rewritten to be more informational than a warning, maybe also the code cleaned up a bit. (Maybe printing all keys and values to be more useful and allow for easy manual fixing afterwards.)

But if you think it is really unnecessary and want the code as short as possible I'm happy to remove it.